native/cocos/renderer/pipeline/GlobalDescriptorSetManager.cpp

```diff
@@ -111,11 +112,12 @@ void GlobalDSManager::update() {
     }
 }
 
-gfx::DescriptorSet *GlobalDSManager::getOrCreateDescriptorSet(uint32_t idx) {
+gfx::DescriptorSet *GlobalDSManager::getOrCreateDescriptorSet(const scene::Light *light) {
+    CC_ASSERT(light);
     // The global descriptorSet is managed by the pipeline and binds the buffer
-    if (_descriptorSetMap.count(idx) <= 0 || (_descriptorSetMap.at(idx) == nullptr)) {
+    if (_descriptorSetMap.count(light) == 0 || (_descriptorSetMap.at(light) == nullptr)) {
         auto *descriptorSet = _device->createDescriptorSet({_descriptorSetLayout});
-        _descriptorSetMap.emplace(idx, descriptorSet);
+        _descriptorSetMap[light] =  descriptorSet;
```

Comment:
Since light will never be removed from `_descriptorSetMap`, will it cause memory leak?

Reply:
it will cause memory leak, then we want fix memory leak on the next issue:

Reply:
light is a raw pointer, will not cause memory leak. The memory leak is these DS are not deleted until pipeline is destroyed.

Reply:
If you mean the item recorded in map, then yep, the item is always exists.

Reply:
Yes, I mean the descriptorSet.